postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -360,11 +367,19 @@ def _run_explain(self, dbname, statement, obfuscated_statement):
                 return None
             return result[0][0]
 
-    def _run_explain_safe(self, dbname, statement, obfuscated_statement):
-        # type: (str, str, str) -> Tuple[Optional[Dict], Optional[DBExplainError], Optional[Exception]]
+    def _run_explain_safe(self, max_query_size, dbname, statement, obfuscated_statement):
+        # type: (int, str, str, str) -> Tuple[Optional[Dict], Optional[DBExplainError], Optional[Exception]]
         if not self._can_explain_statement(obfuscated_statement):
             return None, DBExplainError.no_plans_possible, None
 
+        if self._is_statement_truncated(max_query_size, statement):
+            self._check.count(
+                "dd.postgres.statement_samples.error",
+                1,
+                tags=self._dbtags(dbname, "error:explain-{}".format(DBExplainError.statement_truncated)),
+            )
+            return None, DBExplainError.statement_truncated, None
```

Comment:
Does this mean that in the odd case that a statement is exactly the max length we'll skip collecting an execution plan for it when we could have collected one?

Reply:
I'm wondering if it would be better to always try to explain the statement, and only report as an error if the explain fails.